lib/cucumber/support_code/step_definition.js

```diff
@@ -106,19 +106,30 @@ function StepDefinition(pattern, options, code) {
 
       Cucumber.Util.Exception.registerUncaughtExceptionHandler(handleException, stepDomain);
 
+      var validCodeLengths = self.validCodeLengths(parameters);
+      if (validCodeLengths.indexOf(code.length) === -1) {
+        var error = 'definition has ' + code.length + ' arguments' +
```

Comment:
It'd be better to instantiate an Error object instead of a simple string, imo. 

Also, especially for cucumber beginners, it would be nice to be more explicit about the two options. *"definition has 3 arguments, should have 1 (or 2 if accepting a callback)", something along those lines.